magnolia-voting-app/src/main/java/com/example/votingapp/PollStatusValueProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.example.votingapp;
+
+import java.util.Calendar;
+
+import javax.jcr.*;
+
+import com.vaadin.data.ValueProvider;
+
+/**
+ * PollStatusColumnDefinition
+ */
+public class PollStatusValueProvider implements ValueProvider<Node, String> {
+    
+    @Override
+    public String apply(Node source) {
+        try {
+            if (source.hasProperties()) {
+                Calendar today = Calendar.getInstance();
+                Calendar startDate = source.getProperty("startDate").getValue().getDate();
+                Calendar expiryDate = source.getProperty("expiryDate").getValue().getDate();
+
+                int today_startDate_comparison_value = today.compareTo(startDate);
+                int today_expiryDate_comparison_value = today.compareTo(expiryDate);
+
+                if (today_startDate_comparison_value < 0) {
+                    return "To-do";
+                }
+                if (today_startDate_comparison_value >= 0 && today_expiryDate_comparison_value < 0) {
+                    return "On-going";
+                } 
+                return "Done";
+            }
+        } catch (Exception e) {
```

Comment:
Wrap this in runtime exception. Maybe RuntimeRepositoryException.

If status is not belong to predefined, throw exception immediately, don't return "default"

Reply:
Use logger.error in catch

Reply:
I have replaced Exception by RepositoryException and added logger.